jfix-zookeeper/src/main/kotlin/ru/fix/zookeeper/instance/registry/ServiceInstanceIdRegistry.kt

```
@@ -100,49 +101,34 @@ class ServiceInstanceIdRegistry(
-                            logger.info("ServiceInstanceIdRegistry client reconnected after connection issues " +
-                                    "and got its previous instance id=$instanceId that have before reconnection")
-                        } else {
-                            logger.error("Can't get previous instance id=$instanceId after reconnection." +
-                                    "Lock id: ${Marshaller.marshall(lockIdentity)}. " +
-                                    "Current registration node state: " +
-                                    ZkTreePrinter(curatorFramework).print(config.serviceRegistrationPath, true)
-                            )
-                        }
+            if (lockManager.tryAcquire(lockIdentity) {
+                        logger.error("Failed to prolong lock=$it after reconnect. " +
+                                "Current registration node state: " +
+                                ZkTreePrinter(curatorFramework).print(serviceRegistrationPath, true))
                     }
-                }
-            } catch (e: Exception) {
-                logger.error("Error during reconnection of instance id registry", e)
+            ) {
+                logger.info("ServiceInstanceIdRegistry client reconnected after connection issues " +
+                        "and got its previous instance id=$instanceId that have before reconnection")
+            } else {
+                logger.error("Can't get previous instance id=$instanceId after reconnection. " +
+                        "Probably lock of this instance was expired and new service was registered with this instance id. " +
+                        "Lock id: $lockIdentity. Current registration node state: " +
+                        ZkTreePrinter(curatorFramework).print(serviceRegistrationPath, true)
+                )
             }
+
         }
     }
```

Why you are hiding exception in close Method in catch section?
```
logger.error("Error during instance id registry close.")<- missing exception object in log
```

---

fixed